server/app/category_boards.go

```diff
@@ -1,9 +1,107 @@
 
 func (a *App) GetUserCategoryBoards(userID, teamID string) ([]model.CategoryBoards, error) {
-	return a.store.GetUserCategoryBoards(userID, teamID)
+	categoryBoards, err := a.store.GetUserCategoryBoards(userID, teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	if teamID != "0" {
+		createdCategoryBoards, err := a.createDefaultCategoriesIfRequired(categoryBoards, userID, teamID)
+		if err != nil {
+			return nil, err
+		}
+
+		categoryBoards = append(categoryBoards, createdCategoryBoards...)
+	}
+	return categoryBoards, nil
+}
+
+func (a *App) createDefaultCategoriesIfRequired(existingCategoryBoards []model.CategoryBoards, userID, teamID string) ([]model.CategoryBoards, error) {
+	createdCategories := []model.CategoryBoards{}
+
+	boardsCategoryExist := false
+	for _, categoryBoard := range existingCategoryBoards {
+		if categoryBoard.Name == defaultCategoryBoards {
+			boardsCategoryExist = true
+		}
+	}
+
+	if !boardsCategoryExist {
+		createdCategoryBoards, err := a.createBoardsCategory(userID, teamID, existingCategoryBoards)
+		if err != nil {
+			return nil, err
+		}
+
+		createdCategories = append(createdCategories, *createdCategoryBoards)
+	}
+
+	return createdCategories, nil
+}
+
+func (a *App) createBoardsCategory(userID, teamID string, existingCategoryBoards []model.CategoryBoards) (*model.CategoryBoards, error) {
+	// create the category
+	category := model.Category{
+		Name:      defaultCategoryBoards,
+		UserID:    userID,
+		TeamID:    teamID,
+		Collapsed: false,
+		Type:      model.CategoryTypeSystem,
+	}
+	createdCategory, err := a.CreateCategory(&category)
+	if err != nil {
+		return nil, fmt.Errorf("createBoardsCategory default category creation failed: %w", err)
+	}
+
+	// once the category is created, we need to move all boards which do not
+	// belong to any category, into this category.
+
+	userBoards, err := a.GetBoardsForUserAndTeam(userID, teamID, false)
+	if err != nil {
+		return nil, fmt.Errorf("createBoardsCategory error fetching user's team's boards: %w", err)
+	}
+
+	createdCategoryBoards := &model.CategoryBoards{
+		Category: *createdCategory,
+		BoardIDs: []string{},
+	}
+
+	for _, board := range userBoards {
+		belongsToCategory := false
+
+		for _, categoryBoard := range existingCategoryBoards {
+			for _, boardID := range categoryBoard.BoardIDs {
+				if boardID == board.ID {
+					belongsToCategory = true
+					break
+				}
+			}
+
+			// stop looking into other categories if
+			// the board was found in a category
+			if belongsToCategory {
+				break
+			}
+		}
+
+		if !belongsToCategory {
+			if err := a.AddUpdateUserCategoryBoard(teamID, userID, createdCategory.ID, board.ID); err != nil {
+				return nil, fmt.Errorf("createBoardsCategory failed to add category-less board to the default category, defaultCategoryID: %s, error: %w", createdCategory.ID, err)
+			}
+
+			createdCategoryBoards.BoardIDs = append(createdCategoryBoards.BoardIDs, board.ID)
+		}
+	}
+
+	return createdCategoryBoards, nil
```

Comment:
Couldn't all of this code be eliminated and handle the entire update via SQL.

Reply:
I thought I added a code comment on explaining this, but I might have missed it.

Even if you create the default board categories in SQL migration, you still need the logic to create the categories for new users. Channels does it during the user signup process. We don't have any such thing. So we still need logic in the code to create the default boards category for a new user. Hence, doing everything in code instead of duplicating the logic in code + SQL migration.